### ISSUE TITLE
Quick cell-type selector inconsistency

### ISSUE THREAD

**Issue by Carreau** [jupyter/notebook#805]
The quick cell type selector is a bit inconsistent when you have many cells selected. 

The case is really rare but in the following case:

 - You have various cell type in your selection. 
   - -> the indication on the quick selector is the type of your head cell (issue 1)
 - You want to change your cells type to T (T in code, markdown, raw) 
 - Your Head selection cell is of Type T.
   - > Clicking on  type in quick type selector will not trigger event

--- 

We should document that for now, before trying to come up with a fix that can come later. 
Yes documentation is just a stop-gap but after long discussion online with @fperez it was judge the best on such short notice before release.

**Comment by ellisonbg** [comment]
It should really only show a cell type when all cells selected have the
same type. Otherwise it should be null.

On Tue, Dec 8, 2015 at 11:16 AM, Matthias Bussonnier <
notifications@github.com> wrote:

> The quick cell type selector is a bit inconsistent when you have many
> cells selected.
>
> The case is really rare but in the following case:
>
>    - You have various cell type in your selection.
>       - -> the indication on the quick selector is the type of your head
>       cell (issue 1)
>    - You want to change your cells type to T (T in code, markdown, raw)
>    - Your Head selection cell is of Type T.
>       - > Clicking on type in quick type selector will not trigger event
>
> We should document that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/notebook/issues/805>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by Carreau** [comment]
> It should really only show a cell type when all cells selected have the
> same type. Otherwise it should be null.

The issue was ment to describe the problem, not suggest a fix (could have done a better job in the last sentence). 

What you propose is one of the solution, not the only one. We can discuss that later on, and we already discussed that (with Jon and Fernando) during 4.1 meeting last Tuesday. 

Sorry if it wasn't clear in above message I updated to reflect the status. 

While the solution you propose is close to what might be an ideal solution,
the complexity and edge cases in implementing it now, in comparison to the
rare cases where this inconsistency shows up made us lean toward documenting that this is a known defect instead of hastily fixing it the wrong way.
